chromadb/api/async_client.py

```
@@ -184,11 +186,24 @@ async def create_collection(
     ) -> AsyncCollection:
         if configuration is None:
             configuration = {}
-        if (
-            embedding_function is not None
-            and configuration.get("embedding_function") is None
-        ):
+
+        configuration_ef = configuration.get("embedding_function")
+
+        has_conflict, error_message = has_embedding_function_conflict_on_create(
```

Why not just let this raise? why catch the bool and raise?